### PR TITLE
Firestore: Run 'Firestore Integration Tests' jobs in parallel

### PR DESCRIPTION
**Pull request by dconeybe**
Edit `test-all.yml` to run the `Firestore Integration Tests` job differently. Previously, it called `run_tests_in_ci.js` to run the tests; however, this "wrapper" script has issues where the log output is truncated, making it virtually impossible to debug test failures, or even know _which_ tests are failing. After struggling with a fix for the truncation issue (https://github.com/firebase/firebase-js-sdk/pull/7553) we eventually gave up and instead just create one job per npm script that gets run. This has a few benefits, not the least of which is the tests truly running in parallel but also does _not_ lead to truncated logs.

This PR is heavily based on a similar change to other Firestore github actions checks: https://github.com/firebase/firebase-js-sdk/pull/7523

### CODE REVIEW

File: .github/workflows/test-all.yml [hunk 1]
```diff
@@ -168,12 +172,10 @@ jobs:
       run: |
         cp config/ci.config.json config/project.json
         yarn
-    - name: Set start timestamp env var
-      run: echo "FIREBASE_CI_TEST_START_TIME=$(date +%s)" >> $GITHUB_ENV
     - name: Run unit tests
       run: |
-        xvfb-run yarn lerna run --concurrency 4 test:ci --scope '@firebase/firestore*'
```
Comment (dconeybe) [inline .github/workflows/test-all.yml:175]: NOTE TO SELF: This also needs to test the other packages, like firestore-compat

File: integration/firestore/package.json [hunk 2]
```diff
@@ -6,9 +6,8 @@
     "build:deps": "lerna run --scope @firebase/'{app,firestore}' --include-dependencies build",
     "build:persistence": "INCLUDE_FIRESTORE_PERSISTENCE=true gulp compile-tests",
     "build:memory": "INCLUDE_FIRESTORE_PERSISTENCE=false gulp compile-tests",
+    "karma:singlerun": "karma start --single-run",
     "prettier": "prettier --write '*.js' '*.ts'",
-    "test": "yarn build:memory; karma start --single-run; yarn build:persistence; karma start --single-run;",
```
Comment (wu-hui) [inline integration/firestore/package.json:10]: I think we can maybe keep these or introduce "test:memory" and "test:persistence".

**Review by wu-hui — APPROVED** [review]

**Comment by dconeybe** [comment]
@hsubox76 @DellaBitta Could one of you take a look at this PR? I think it needs your approval to merge.

File: integration/firestore/package.json (hunk 2 above)
Comment (dconeybe) [reply integration/firestore/package.json:10]: Lines 12 and 14 define exactly "test:memory" and "test:persistence"!

Comment (dconeybe) [reply integration/firestore/package.json:10]: I just wanted to clean up unused entries in this "scripts" list to reduce future confusion.

**Review by DellaBitta — APPROVED** [review]

**Pull request merged**